fabric-component-access-api-v1/src/main/java/net/fabricmc/fabric/impl/component/access/ComponentTypeRegistryImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.component.access;
+
+import com.google.common.base.Preconditions;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.component.access.v1.ComponentType;
+import net.fabricmc.fabric.api.component.access.v1.ComponentTypeRegistry;
+
+public class ComponentTypeRegistryImpl implements ComponentTypeRegistry {
```

Comment:
I like final things (this class could be final)

Reply:
changed